src/SFA.DAS.ApprenticeCommitments/Infrastructure/ServiceCollectionExtensions.cs

```diff
@@ -52,20 +53,29 @@ public static IServiceCollection AddEntityFrameworkForApprenticeCommitments(this
             {
                 var unitOfWorkContext = p.GetService<IUnitOfWorkContext>();
                 var connectionFactory = p.GetService<IConnectionFactory>();
+                var loggerFactory = p.GetService<ILoggerFactory>();
 
                 ApprenticeCommitmentsDbContext dbContext;
                 try
                 {
                     var synchronizedStorageSession = unitOfWorkContext.Get<SynchronizedStorageSession>();
                     var sqlStorageSession = synchronizedStorageSession.GetSqlStorageSession();
                     var optionsBuilder = new DbContextOptionsBuilder<ApprenticeCommitmentsDbContext>().UseDataStorage(connectionFactory, sqlStorageSession.Connection);
+                    if (config.IsLocalAcceptanceOrDev())
+                    {
+                        optionsBuilder.EnableSensitiveDataLogging().UseLoggerFactory(loggerFactory);
+                    }
```

Comment:
The duplication is a little obvious now, with the extra branch appearing in both try and catch blocks.  It may be worth extracting the options/context lines into a method?

Reply:
agree

Reply:
The check and `EnableSensitiveDataLogging` isn't needed here any more is it, as it's in the `UseLocalSqlLogger` extension now.